### PR TITLE
upgrade to sf 5.2

### PR DESCRIPTION
**Pull request by HedicGuibert**
This PR upgrades the Symfony version to 5.2

### CODE REVIEW

File: assets/controllers/hello_controller.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Controller } from 'stimulus';
```
Comment (lyrixx) [inline assets/controllers/hello_controller.js:1]: Seems not used ; to remove

File: composer.json [hunk 2]
```diff
@@ -26,40 +26,40 @@
...
         "php-http/guzzle6-adapter": "^2.0",
         "php-http/message": "^1.8",
-        "php-translation/symfony-bundle": "^0.9",
+        "php-translation/symfony-bundle": "^0.12",
         "sameer-shelavale/php-countries-array": "^1.0",
         "sensio/framework-extra-bundle": "^5.2",
         "stof/doctrine-extensions-bundle": "^1.3",
-        "symfony/asset": "^4.4",
-        "symfony/console": "^4.4",
+        "symfony/asset": "^5.0",
```
Comment (lyrixx) [inline composer.json:33]: let's use 5.2; same for other deps

File: config/services.yaml [hunk 3]
```diff
@@ -9,17 +9,22 @@ services:
         App\Fetcher\FetcherInterface:
             tags: ['app.fetcher']
 
+
```
Comment (lyrixx) [inline config/services.yaml:12]: to remove

File: package.json [hunk 4]
```diff
@@ -1,42 +1,18 @@
...
-        "babel-loader": "^7.1.2",
-        "babel-preset-env": "^1.6.1",
-        "babel-preset-es2015": "^6.24.1",
-        "babel-preset-stage-2": "^6.24.1",
-        "copy-webpack-plugin": "^4.2.1",
-        "node-sass": "^4.7.2",
-        "prettier": "1.8.2",
-        "sass-loader": "^6.0.6",
-        "webpack-notifier": "^1.5.0"
+        "@symfony/stimulus-bridge": "^1.1.0",
```
Comment (lyrixx) [inline package.json:3]: Are you using it? I dont thinks so

File: package.json [hunk 5]
```diff
@@ -1,42 +1,18 @@
...
-        "build": "encore production",
-        "check-prettier": "prettier \"assets/js/**/*.js\"",
-        "prettier": "prettier --write \"assets/js/**/*.js\""
-    },
-    "dependencies": {
-        "bootstrap": "^4.3",
-        "font-awesome": "^4.7.0",
-        "jquery": "^3.5.0",
-        "popper.js": "^1.15"
-    },
```
Comment (lyrixx) [inline package.json:32]: This is really suspect

File: package.json [hunk 6]
```diff
@@ -1,42 +1,18 @@
...
-    "prettier": {
-        "parser": "flow",
-        "bracketSpacing": true,
-        "singleQuote": true,
-        "jsxBracketSameLine": true,
-        "trailing-comma": "all",
-        "printWidth": 100,
-        "tabWidth": 4
```
Comment (lyrixx) [inline package.json:40]: This too !

File: package.json [hunk 7]
```diff
@@ -1,42 +1,18 @@
 {
-    "name": "starfleet",
-    "version": "0.0.1",
-    "license": "MIT",
-    "private": true,
```
Comment (lyrixx) [inline package.json:5]: Why did you remove it?

File: src/Command/InitCommand.php [hunk 8]
```diff
@@ -41,8 +41,7 @@ protected function configure()
             ->setDescription('Init command that create first admin user.')
             ->addArgument('name', InputArgument::REQUIRED, 'Admin name')
             ->addArgument('email', InputArgument::REQUIRED, 'Admin email')
-            ->addArgument('password', InputArgument::REQUIRED, 'Admin password')
-        ;
+            ->addArgument('password', InputArgument::REQUIRED, 'Admin password');
```
Comment (lyrixx) [inline src/Command/InitCommand.php:44]: The previous version was better

File: src/DataFixtures/AppFixtures.php [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+use Doctrine\Persistence\ObjectManager;
+
+class AppFixtures extends Fixture
+{
+    public function load(ObjectManager $manager)
+    {
+        // $product = new Product();
+        // $manager->persist($product);
+
+        $manager->flush();
```
Comment (lyrixx) [inline src/DataFixtures/AppFixtures.php:24]: This file is not used. To remove

File: src/Repository/ConferenceRepository.php [hunk 10]
```diff
@@ -38,16 +38,14 @@ public function findEndingCfps(): array
             ->setParameter('today', $today)
             ->setParameter('threshold', $threshold)
             ->getQuery()
-            ->execute()
-        ;
+            ->execute();
```
Comment (lyrixx) [inline src/Repository/ConferenceRepository.php:41]: Previous version was better. Same for other similar diff

File: .circleci/config.yml [hunk 11]
```diff
@@ -40,7 +40,7 @@ jobs:
                   command: composer install --no-scripts
             - run:
                   name: Create .env
-                  command: cp .env.dist .env
+                  command: cp .env
```
Comment (lyrixx) [inline .circleci/config.yml]: Does not make sens (and does not work I guess)

File: assets/controllers/hello_controller.js [hunk 12]
```diff
@@ -1,5 +1,3 @@
-import { Controller } from 'stimulus';
```
Comment (lyrixx) [inline assets/controllers/hello_controller.js]: this file is not used at all, not only the import... 😬

File: package.json [hunk 13]
```diff
@@ -1,18 +1,39 @@
...
+        "singleQuote": true,
+        "jsxBracketSameLine": true,
+        "trailing-comma": "all",
+        "printWidth": 100,
+        "tabWidth": 4,
         "build": "encore production --progress"
     }
 }
+
+
```
Comment (lyrixx) [inline package.json:39]: Nope

File: .circleci/config.yml (hunk 11 above)
Comment (HedicGuibert) [reply .circleci/config.yml]: Yes I completely removed it later; it still doesn't work though :stuck_out_tongue:

File: .env [hunk 14]
```diff
@@ -13,14 +9,15 @@ ROUTER_HOST=local.starfleet.app
 
 ###> symfony/framework-bundle ###
 APP_ENV=dev
-APP_SECRET=e405abac97d81003cfc9c233b8995a73
-TRUSTED_HOSTS=local.starfleet.app
```
Comment (lyrixx) [inline .env:17]: Where does this configuration went?

File: assets/app.js [hunk 15]
```diff
@@ -0,0 +1,12 @@
+/*
+ * Welcome to your app's main JavaScript file!
+ *
+ * We recommend including the built version of this JavaScript file
+ * (and its CSS file) in your base layout (base.html.twig).
+ */
+
+// any CSS you import will output into a single css file (app.css in this case)
```
Comment (lyrixx) [inline assets/app.js:8]: Useless, to drop

File: assets/app.js [hunk 16]
```diff
@@ -0,0 +1,12 @@
...
+import './styles/app.css';
+
+// start the Stimulus application
```
Comment (lyrixx) [inline assets/app.js:11]: I don't thinks so

File: assets/bootstrap.js [hunk 17]
```diff
@@ -0,0 +1,5 @@
+import { startStimulusApp } from '@symfony/stimulus-bridge';
+import '@symfony/autoimport';
+
+// Registers Stimulus controllers from controllers.json and in the controllers/ directory
+export const app = startStimulusApp(require.context('./controllers', true, /\.(j|t)sx?$/));
```
Comment (lyrixx) [inline assets/bootstrap.js:5]: Are you using stimulus? (It's the 3rd time I'm asking this question). I don't think so. 
Please remove all code related to symfony/ux. It's not used in this application.

File: assets/controllers.json [hunk 18]
```diff
@@ -0,0 +1,4 @@
+{
+    "controllers": [],
```
Comment (lyrixx) [inline assets/controllers.json:2]: File not used

File: assets/styles/app.css [hunk 19]
```diff
@@ -0,0 +1,3 @@
+body {
+    background-color: lightgray;
```
Comment (lyrixx) [inline assets/styles/app.css:2]: Where does it come from? It was not here before upgrading to sf 5.x I think

File: composer.json [hunk 20]
```diff
@@ -26,40 +26,40 @@
...
     "require-dev": {
         "doctrine/doctrine-fixtures-bundle": "^3.0",
         "friendsofphp/php-cs-fixer": "^2.9",
-        "symfony/browser-kit": "^4.4",
-        "symfony/css-selector": "^4.4",
-        "symfony/phpunit-bridge": "^4.4",
+        "symfony/browser-kit": "^5.2",
+        "symfony/css-selector": "^5.2",
+        "symfony/phpunit-bridge": "^5.2",
         "symfony/thanks": "^1.0"
```
Comment (lyrixx) [inline composer.json:63]: This file miss :

```
    "extra": {
        "symfony": {
            "require": "5.2.*"
        }
    }
```

And the ID is useless

File: config/packages/doctrine_migrations.yaml [hunk 21]
```diff
@@ -1,5 +1,5 @@
 doctrine_migrations:
-    dir_name: '%kernel.project_dir%/src/Migrations'
-    # namespace is arbitrary but should be different from App\Migrations
-    # as migrations classes should NOT be autoloaded
-    namespace: DoctrineMigrations
+    migrations_paths:
+        # namespace is arbitrary but should be different from App\Migrations
+        # as migrations classes should NOT be autoloaded
+        'DoctrineMigrations': '%kernel.project_dir%/migrations'
```
Comment (lyrixx) [inline config/packages/doctrine_migrations.yaml:5]: Please force doctrine migration 2, to be able to use this recipies

File: package.json [hunk 22]
```diff
@@ -1,21 +1,25 @@
...
-        "webpack-notifier": "^1.5.0"
+        "regenerator-runtime": "^0.13.2",
+        "sass-loader": "^9.0.1",
+        "stimulus": "^2.0.0",
+        "webpack-notifier": "^1.6.0"
     },
+    "name": "starfleet",
+    "version": "0.0.1",
+    "license": "MIT",
+    "private": true,
```
Comment (lyrixx) [inline package.json:22]: This should be at the top of the document + Please avoid creating useless diff (code deplacement bring nothing)

File: package.json (hunk 4 above)
Comment (lyrixx) [reply package.json:3]: I'll rephrase: you are not really using, and you must not use it.

I unresolve the issue

File: package.json [hunk 23]
```diff
@@ -1,21 +1,25 @@
...
         "babel-preset-stage-2": "^6.24.1",
         "copy-webpack-plugin": "^4.2.1",
+        "core-js": "^3.0.0",
         "node-sass": "^4.7.2",
         "prettier": "1.8.2",
-        "sass-loader": "^6.0.6",
-        "webpack-notifier": "^1.5.0"
+        "regenerator-runtime": "^0.13.2",
+        "sass-loader": "^9.0.1",
+        "stimulus": "^2.0.0",
```
Comment (lyrixx) [inline package.json:16]: To remove

File: package.json [hunk 24]
```diff
@@ -1,21 +1,25 @@
...
         "babel-preset-env": "^1.6.1",
         "babel-preset-es2015": "^6.24.1",
         "babel-preset-stage-2": "^6.24.1",
         "copy-webpack-plugin": "^4.2.1",
+        "core-js": "^3.0.0",
         "node-sass": "^4.7.2",
         "prettier": "1.8.2",
-        "sass-loader": "^6.0.6",
-        "webpack-notifier": "^1.5.0"
+        "regenerator-runtime": "^0.13.2",
```
Comment (lyrixx) [inline package.json:14]: What is it?

File: package.json [hunk 25]
```diff
@@ -1,21 +1,25 @@
...
-        "@symfony/webpack-encore": "^0.17.0",
+        "@symfony/stimulus-bridge": "^1.1.0",
+        "@symfony/webpack-encore": "^0.32.0",
         "babel-core": "^6.26.0",
         "babel-loader": "^7.1.2",
         "babel-preset-env": "^1.6.1",
         "babel-preset-es2015": "^6.24.1",
         "babel-preset-stage-2": "^6.24.1",
         "copy-webpack-plugin": "^4.2.1",
+        "core-js": "^3.0.0",
```
Comment (lyrixx) [inline package.json:11]: Why do we need that?

File: package.json [hunk 26]
```diff
@@ -37,6 +41,7 @@
         "jsxBracketSameLine": true,
         "trailing-comma": "all",
         "printWidth": 100,
-        "tabWidth": 4
+        "tabWidth": 4,
+        "build": "encore production --progress"
```
Comment (lyrixx) [inline package.json:45]: This is suspect

File: src/Kernel.php [hunk 27]
```diff
@@ -12,59 +12,36 @@
...
-            if (isset($envs['all']) || isset($envs[$this->environment])) {
-                yield new $class();
-            }
+        $container->import('../config/{packages}/*.yaml');
+        $container->import('../config/{packages}/'.$this->environment.'/*.yaml');
+
+        if (is_file(\dirname(__DIR__).'/config/services.yaml')) {
+            $container->import('../config/services.yaml');
+            $container->import('../config/{services}_'.$this->environment.'.yaml');
+        } elseif (is_file($path = \dirname(__DIR__).'/config/services.php')) {
```
Comment (lyrixx) [inline src/Kernel.php:31]: all your service are in YAML, not PHP, so you can drop this condition and block

File: src/Kernel.php [hunk 28]
```diff
@@ -12,59 +12,36 @@
...
-    {
-        $confDir = $this->getProjectDir().'/config';
-        if (is_dir($confDir.'/routes/')) {
-            $routes->import($confDir.'/routes/*'.self::CONFIG_EXTS, '/', 'glob');
-        }
-        if (is_dir($confDir.'/routes/'.$this->environment)) {
-            $routes->import($confDir.'/routes/'.$this->environment.'/**/*'.self::CONFIG_EXTS, '/', 'glob');
+        if (is_file(\dirname(__DIR__).'/config/routes.yaml')) {
+            $routes->import('../config/routes.yaml');
+        } elseif (is_file($path = \dirname(__DIR__).'/config/routes.php')) {
```
Comment (lyrixx) [inline src/Kernel.php:43]: Same here

File: webpack.config.js [hunk 29]
```diff
@@ -1,31 +1,37 @@
...
-    .configureBabel(function(babelConfig) {
-        babelConfig.presets.push('es2015');
+    .enableSourceMaps(!Encore.isProduction())
+    .enableVersioning(Encore.isProduction())
+    .configureBabel((config) => {
+        config.plugins.push('@babel/plugin-proposal-class-properties');
+    })
+    .configureBabelPresetEnv((config) => {
+        config.useBuiltIns = 'usage';
+        config.corejs = 3;
```
Comment (lyrixx) [inline webpack.config.js:24]: This diff is messy and IMHO it does not bring anything usefull to re-ordonate line.


Plus there are suspect things
*  ```diff
    -     .addEntry('js/app', './assets/js/app.js')
    +    .addEntry('app', './assets/app.js')
    ```
    IMHO, the front is not going to work anymore

* stimulus should not be used

* `splitEntryChunks()` was not used, please remove it

* `enableSingleRuntimeChunk()` was not used, please remove it

* I'm not sure about babel conf, let's do that in another PR if needed

---

This PR is about updating SF (PHP), not the JS stack. It may be legit to update webpack encore, since it's manager by symfony. but other things must stay identical to `master`

File: .env [hunk 30]
```diff
@@ -10,6 +10,7 @@ ROUTER_HOST=local.starfleet.app
 ###> symfony/framework-bundle ###
 APP_ENV=dev
 APP_SECRET=7e2a14a401d119578cf7db2cba26030e
+TRUSTED_HOSTS=local.starfleet.app
```
Comment (lyrixx) [inline .env:13]: Where do you use this var?

File: src/Kernel.php [hunk 31]
```diff
@@ -28,8 +28,6 @@ protected function configureContainer(ContainerConfigurator $container): void
         if (is_file(\dirname(__DIR__).'/config/services.yaml')) {
```
Comment (lyrixx) [inline src/Kernel.php:28]: The file do exist. Not need for a if, same after

**Review by lyrixx — COMMENTED** [review]
Nice, only one comment and we'll be good.

@welcoMattic if you want to review it, it's now :)

File: webpack.config.js [hunk 32]
```diff
@@ -1,31 +1,35 @@
...
```
Comment (welcoMattic) [inline webpack.config.js:22]: I trust you here, I'm not aware of the babel config last updates

**Review by welcoMattic — APPROVED** [review]
LGTM :clap:

**Review by lyrixx — APPROVED** [review]
Very nice 👍🏼

**Comment by lyrixx** [comment]
Here we go 🎉 thanks
